base/loggers/OwnSplitChannel.cpp

```diff
@@ -26,16 +27,52 @@ void OwnSplitChannel::log(const Poco::Message & msg)
         auto matches = masker->wipeSensitiveData(message_text);
         if (matches > 0)
         {
-            logSplit({msg, message_text}); // we will continue with the copy of original message with text modified
+            tryLogSplit({msg, message_text}); // we will continue with the copy of original message with text modified
             return;
         }
 
     }
 
-    logSplit(msg);
+    tryLogSplit(msg);
 }
 
 
+void OwnSplitChannel::tryLogSplit(const Poco::Message & msg)
+{
+    try
+    {
+        logSplit(msg);
+    }
+    catch (...)
+    {
+        /// It is better to catch the errors here in order to avoid
+        /// breaking some functionality because of unexpected "File not
+        /// found" (or similar) error.
+        ///
+        /// For example StorageDistributedDirectoryMonitor will mark batch
+        /// as broken, some MergeTree code can also be affected.
+        ///
+        /// Also note, that we cannot log the exception here, since this
+        /// will lead to recursion, using regular tryLogCurrentException().
+        /// but let's log it into the stderr at least.
+        std::string message = "Cannot add message to the log:\n";
```

Comment:
But these lines also can lead to exceptions, for example, `bad_alloc`. However, it should be better to throw them than silently catch and ignore them....

Reply:
Right, I though that logging uses `LockExceptionInThread`/`BlockerInThread` but it does not.

To me it sounds like it just should use `LockExceptionInThread`, since memory limit error during logging will hide log message (plus it is blocked in few places already, like `SystemLog<>::add`) @alesapin thoughts?

Reply:
>To me it sounds like it just should use LockExceptionInThread,

Done